ACE/ace/config-integrity-common.h

```diff
@@ -0,0 +1,185 @@
+#endif
+#define ACE_HAS_DIRENT
+
+#define ACE_PAGE_SIZE 4096
+
+#define ACE_LACKS_KEY_T
+
+#define ACE_LACKS_ISCTYPE
+#define ACE_LACKS_WCSICMP
+#define ACE_LACKS_WCSNICMP
+#define ACE_LACKS_WCSDUP
+#define ACE_LACKS_WCSTOK
+#define ACE_LACKS_ALPHASORT
+#define ACE_LACKS_ITOW
+
+#define ACE_HAS_SIGWAIT
+#define ACE_HAS_PTHREAD_SIGMASK_PROTO
+
+#define ACE_LACKS_SUSECONDS_T
+#define ACE_LACKS_USECONDS_T
+#define ACE_HAS_POSIX_TIME
+#define ACE_HAS_STRDUP_EMULATION
+#define ACE_HAS_MSG
+#define ACE_LACKS_CONDATTR_PSHARED
+#define ACE_LACKS_MUTEXATTR_PSHARED
+#define ACE_LACKS_RLIMIT
+#define ACE_LACKS_COND_T
+#define ACE_LACKS_RWLOCK_T
+#define ACE_LACKS_SEMBUF_T
+#define ACE_LACKS_UNIX_DOMAIN_SOCKETS
+#define ACE_LACKS_NETDB_REENTRANT_FUNCTIONS
+#define ACE_LACKS_PIPE
+#define ACE_LACKS_SYS_PARAM_H
+#define ACE_LACKS_SYS_MSG_H
+#define ACE_LACKS_PWD_FUNCTIONS
+#define ACE_LACKS_UNIX_SIGNALS
+#define ACE_LACKS_SYSV_SHMEM
+#define ACE_LACKS_PUTENV
+#define ACE_HAS_4_4BSD_SENDMSG_RECVMSG
+#define ACE_LACKS_THREAD_PROCESS_SCOPING
+#define ACE_LACKS_SETSCHED
+#define ACE_LACKS_STRRECVFD
+#define ACE_LACKS_SYSCONF
+
+#define ACE_LACKS_REALPATH
+#define ACE_LACKS_SWAB
+#define ACE_HAS_CONSISTENT_SIGNAL_PROTOTYPES
+
+/***** Not tied to standards AFAIK ****/
+#define ACE_LACKS_MADVISE /* paging optimization not needed with INTEGRITY */
+#define ACE_LACKS_INTTYPES_H
+#define ACE_LACKS_SYS_RESOURCE_H
+#define ACE_LACKS_SEARCH_H
+#define ACE_LACKS_SYS_IPC_H
+#define ACE_LACKS_SYS_SEM_H
+#define ACE_LACKS_STROPTS_H
+#define ACE_LACKS_SYS_SHM_H
+#define ACE_LACKS_MEMORY_H
+
+#define ACE_LACKS_IOSTREAM_TOTALLY
+
+// Add the following macros to config.h to
+// build with --no-exceptions option in MULTI IDE
+//#define ACE_INTEGRITY_NO_EXCEPT
+
+#define TAO_USE_DOTTED_DECIMAL_ADDRESSES 1
+
+#include <INTEGRITY.h>
+
+#include <time.h>
+
+typedef void (*__sighandler_t) (int);
+
+#ifdef ppc
+# define ACE_HAS_POWERPC_TIMER
+#endif
+
+#define ACE_HAS_NONCONST_SELECT_TIMEVAL
+#include <sys/uio.h> // needed to define iovec
+#define ACE_LACKS_READLINK
+#define ACE_LACKS_GETPPID
+#define ACE_USES_STD_NAMESPACE_FOR_STDCPP_LIB 1
+#define ACE_LACKS_CUSERID
+
+#define ACE_DEFAULT_THREAD_PRIORITY 127
+#define PRI_FIFO_MIN 1
+#define PRI_FIFO_MAX 127
+#define ACE_THR_PRI_FIFO_DEF 127
+#define PRI_RR_MIN 1
+#define PRI_RR_MAX 127
+#define ACE_THR_PRI_RR_DEF 127
+#define PRI_OTHER_MIN 1
+#define PRI_OTHER_MAX 127
+#define ACE_THR_PRI_OTHER_DEF 127
+#define ACE_PTHREAD_RETURN_ON_EXIT
+#undef ACE_LACKS_UNLINK
+
+#define ACE_HAS_TIMED_MESSAGE_BLOCKS
+
+extern "C" {
+  int unlink (const char *);
+}
+
+#define ACE_LACKS_UNIX_SYSLOG
+#define ACE_LACKS_TELLDIR
+#define ACE_LACKS_SEEKDIR
+#define ACE_LACKS_GETHOSTENT
+
+#if !_HAS_EXCEPTIONS
+# define ACE_STD_ALLOCATOR_NOTHROW
+#endif
+
+// When config.h defines ACE_NO_GHS_LIBIO, the functions provided by the
+// INTEGRITY-178 Low-Level I/O Library (libio_178b_be.a) are not called by ACE.
+#ifdef ACE_NO_GHS_LIBIO
```

Comment:
Why not used ACE_NO_INTEGRITY_LIBIO, sometimes you use ghs, any reason to not use integrity?

Reply:
I think this macro is used to exclude a few I/O functions from both INTEGRITY and INTEGRITY-178, thus it's named with _GHS_ in it.